### DISCUSSION TITLE
Div not updating in Grid Event

### DISCUSSION THREAD

**Discussion by pjk633** [elimintz/justpy#650] [Q&A]
Hello,

New to justpy but loving what I've seen so far. I've been playing around with the Grid Events example 1 tutorial, plugging in my own data source from an MS SQL Server database and I think I may have uncovered a possible issue. In the code below the Div gets updated exactly as it should as long as I retrieve no more than 340 rows from this particular database. If I try to retrieve more, although the event gets triggered and row_data_div.text is assigned the correct value, this isn't reflected on the webpage and the Div continues to display just 'Testing'.

Any ideas? 

Phil
```python
import justpy as jp
import pandas as pd
from sqlalchemy.engine import URL
from sqlalchemy import create_engine

connection_string = "Driver={ODBC Driver 17 for SQL Server}; Server=(localdb)\DigitalText;Database=Menota; Trusted_Connection=yes"
connection_url = URL.create("mssql+pyodbc", query={"odbc_connect": connection_string})

engine = create_engine(connection_url)

wm_df = pd.read_sql_query("SELECT TOP 340 Transform.*, sub_query.Occurs \
                            FROM Transform \
                            LEFT OUTER JOIN ( \
                                SELECT Manuscript.Facs, COUNT(*) AS Occurs \
                                FROM Manuscript \
                                GROUP BY Manuscript.Facs) sub_query \
                            ON Transform.Facs = sub_query.Facs", engine)                            

def row_selected2(self, msg):
    if msg.selected:
        self.row_data_div.text = msg.data
        self.row_selected2 = msg.rowIndex
    elif self.row_selected2 == msg.rowIndex:
        self.row_data_div.text = ''

        
def db_test():
    wp = jp.WebPage()
    row_data_div = jp.Div(text='Testing', a=wp)
    g = jp.AgGrid(a=wp)
    g.load_pandas_frame(wm_df)
    g.options.domLayout='autoHeight'                           
    g.options.pagination = True
    g.options.paginationPageSize = 15
    g.options.columnDefs[0].checkboxSelection = True
    g.options.columnDefs[4].editable = True
    g.options.columnDefs[4].cellEditor = 'agSelectCellEditor'
    g.options.columnDefs[4].cellEditorParams = {'values': ['TestA', 'TestB']}
    g.row_data_div = row_data_div
    g.on('rowSelected', row_selected2)

    return wp

jp.justpy(db_test)
```
Just to add, I have since tested this using an identical table in sqlite and the result is exactly the same – works fine if I limit the number of rows returned to 340, but any more than that and the Div element stops being updated on the page.

**Comment by WolfgangFahl** [comment]
**Any ideas?**
Yes - row 341 might contain data that aggrid doesn't like in the way it is converted by load_pandas_frame see #634 with a similar situation where we already asked for the  concrete data causing the problem. You might want to convert you data to a list of dicts and try out the dict of the  single row that i would assume is the culprit. Then we can create an issue similar to #500 and #270 and fix the problem.

**Comment by pjk633** [comment]
Thanks for the reply. Yes, it turns out row 341 (as also some other rows later in the dataset) contained a field set to NULL as a result of the LEFT OUTER JOIN. I can get round that easily by using ifnull in the SELECT statement.

What threw me though, is that the aggrid itself works perfectly even with the NULL values, it's the link between the Aggrid property used to store the jp.Div on the webpage (as referenced by the rowselected event) that seems to break.
